llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -5152,7 +5152,8 @@ ElementCount LoopVectorizationCostModel::getMaximizedVFForTarget(
                            : TargetTransformInfo::RGK_FixedWidthVector;
   ElementCount MaxVF = MaxVectorElementCount;
   if (MaximizeBandwidth || (MaximizeBandwidth.getNumOccurrences() == 0 &&
-                            TTI.shouldMaximizeVectorBandwidth(RegKind))) {
+                            (TTI.shouldMaximizeVectorBandwidth(RegKind) ||
+                             Legal->hasVectorVariants()))) {
```

Comment:
If you agree with my suggestions above you can rewrite this as:

TTI.shouldMaximizeVectorBandwidth(RegKind) ||
 (UseWiderVFIfVariantsPresent && Legal->hasVectorVariants())